### PR TITLE
Add an option to choose AWS SES for email

### PR DESCRIPTION
**Pull request by joshtensibly**
[//]: # (Thank you for helping us out: your efforts mean great deal to the project and the community as a whole!)

[//]: # (Before you proceed:)

[//]: # (1. Make sure to add yourself to `CONTRIBUTORS.rst` through this PR provided you're contributing here for the first time)
[//]: # (2. Don't forget to update the `docs/` presuming others would benefit from a concise description of whatever that you're proposing)

## Description

[//]: # (What's it you're proposing?)
Add the option to choose SES as email provider. This will install `django-anymail[amazon_ses]` and set the `EMAIL_BACKEND` to `anymail.backends.amazon_ses.EmailBackend`

## Rationale

[//]: # (Why does the project need that?)
To allow users to start a project with SES for email instead of Mailgun of they wish.

## Use case(s) / visualization(s)
Send emails with SES.
[//]: # ("Better to see something once than to hear about it a thousand times.")

### CODE REVIEW

**Comment by foarsitter** [comment]
When you select a different value for `cloud_provider` then `AWS`, the `AWS_*` variables are not added to the settings. This breaks you configuration for SES since the credentials are not provided.  

https://github.com/pydanny/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/.envs/.production/.django#L19

File: cookiecutter.json [hunk 1]
```diff
@@ -17,22 +17,10 @@
...
-    "Gulp"
-  ],
-  "cloud_provider": [
-    "AWS",
-    "GCP",
-    "None"
-  ],
+  "postgresql_version": ["11.3", "10.8", "9.6", "9.5", "9.4"],
+  "js_task_runner": ["None", "Gulp"],
+  "cloud_provider": ["AWS", "GCP", "None"],
```
Comment (browniebroke) [inline cookiecutter.json]: Could you please not change the lines wrapping and stick to the code style of the file? We tend to have 1 option per line as it's making diff easier to process.

File: CONTRIBUTORS.rst [hunk 2]
```diff
@@ -160,6 +160,7 @@ Listed in alphabetical order.
   Meghan Heintz            `@dot2dotseurat`_
   Mesut Yılmaz             `@myilmaz`_
   Michael Gecht            `@mimischi`_                  @_mischi
+  Michael Karamuth         `@joshtensibly`_
```
Comment (browniebroke) [inline CONTRIBUTORS.rst]: This is creating a broken link, we need to add the target for `@joshtensibly` firther down in the file.

**Comment by browniebroke** [comment]
Thanks for your effort! I forgot about this, but it's now superseded by #2435, closing

**Pull request closed without merging**